src/differentiation/ode_callback.jl

```diff
@@ -0,0 +1,211 @@
+
+"""
+    maybe_switch_chart(u, t, integrator)
+
+Terminate integration when integrator goes too closely to chart boundary.
+"""
+function maybe_switch_chart(u, t, integrator)
```

Comment:
this functions name sounds strange compared to what the documentation says

for me a suitable name would be `terminate_integrator_near_chart_boundary`?

Could you maybe also document the parameters? Also the chart switch has a boundary parameter, we could pass that one on?

Reply:
`terminate_integrator_near_chart_boundary` sounds better, thanks for the suggestion. I will expand it a bit to include that boundary parameter here.

Reply:
oh and it has to be `determinate` I think.

Reply:
What has to be `determinate`?

Reply:
The function name should be `determinate_integrator_near_chart_boundary` since `terminate` would mean to end (and not to find out).

Reply:
Well, but it actually terminates the internal integrator?

Reply:
Oh, me re-reading PRs before first coffee 😴